### PR TITLE
Remove MP11 dependency

### PR DESCRIPTION
**Pull request by mborland**
Removes the only MP11 dependency. Does not need to make it in for release.

### CODE REVIEW

File: include/boost/math/tools/mp.hpp [hunk 1]
```diff
@@ -335,6 +335,108 @@ using mp_remove_if = typename detail::mp_remove_if_impl<L, P>::type;
 template<typename L, typename Q> 
 using mp_remove_if_q = mp_remove_if<L, Q::template fn>;
 
+// Index sequence
+// Use C++14 index sequence if available
+#ifndef BOOST_NO_CXX14_VARIABLE_TEMPLATES
+#include <utility>
+template<std::size_t... I>
+using index_sequence = std::index_sequence<I...>;
```
Comment (mborland) [inline include/boost/math/tools/mp.hpp:343]: @jzmaddock `std::index_sequence` is a C++14 feature, but I didn't see anything in config that specifically called it out. Is there something better to put here?

Comment (jzmaddock) [reply include/boost/math/tools/mp.hpp:343]: Try `defined(__cpp_lib_integer_sequence) && (__cpp_lib_integer_sequence >= 201304)`

Comment (mborland) [reply include/boost/math/tools/mp.hpp:343]: That does the trick locally, hopefully CI agrees. Thanks.

**Comment by mborland** [comment]
@jzmaddock This should be good to go.

**Comment by jwakely** [comment]
This is badly broken, see #670

**Pull request merged**